SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp

```diff
@@ -256,11 +265,65 @@ void ASpatialTestRepNotify::PrepareTest()
 			FinishStep();
 		},
 		nullptr, 5.0f);
+
+	/**
+	 * In this section of the test, we check the order rep notifies are called in.
+	 * In native, data will be applied for an actor, its subobjects, and only THEN will repnotifies be called for the updated data.
+	 *
+	 * To test this, we set the update a replicated property on an actor (the test itself) and a subobject.
+	 * Then in repnotifies on both the actor and the subobject, we check that the property on the other object has already been updated.
+	 * */
+
+	AddStep(
+		TEXT("SpatialTestRepNotifyAllWorkersInitialiseExpectedOrderingProps"), FWorkerDefinition::AllWorkers, nullptr,
+		[this]() {
+			AssertIsValid(TestSubobject, TEXT("TestSubobject should be valid."));
+			TestSubobject->ExpectedParentInt1Property = 350;
+			TestSubobject->bParentPropertyWasExpectedProperty = false;
+			ExpectedSubobjectIntProperty = 400;
+			bSubobjectIntPropertyWasExpectedProperty = false;
+			FinishStep();
+		},
+		nullptr);
+
+	AddStep(
+		TEXT("SpatialTestRepNotifyServerChangeReplicatedVariables2"), FWorkerDefinition::Server(1), nullptr,
+		[this]() {
+			AssertIsValid(TestSubobject, TEXT("TestSubobject should be valid."));
+			TestSubobject->OnChangedRepNotifyInt = 400;
+			OnChangedRepNotifyInt1 = 350;
+			FinishStep();
+		},
+		nullptr);
+
+	AddStep(
+		TEXT("SpatialTestRepNotifyClientCheckOrderingWasCorrect"), FWorkerDefinition::AllClients,
+		[this]() -> bool {
+			return TestSubobject->OnChangedRepNotifyInt == 400 && OnChangedRepNotifyInt1 == 350;
+		},
```

Comment:
I would slightly prefer `if (RequireEqual()...) {the other stuff}`
but mostly a nit

Reply:
ah you can use requires etc in isreadyevent, nice. I've changed to requires then return true, I guess that's how it's done

Reply:
oh, sorry, should have explained more
I wanted to move everything to a tick function with Requires
Not actually sure if requires work in the ready event, though they may...

Reply:
hmm, well the test is still passing, but will move across